tests/src/python/test_provider_virtual.py

```diff
@@ -1271,6 +1271,27 @@ def test_subset_string(self):
         self.assertEqual(gpkg_virtual_layer.featureCount(), 1)
         self.assertEqual(gpkg_virtual_layer.subsetString(), '"join_value" = \'twenty\'')
 
+    def test_feature_count_on_error(self):
+
```

Comment:
```suggestion
        """Test that triggered exception while getting feature count on a badly defined
        virtual layer is correctly caught (see https://github.com/qgis/QGIS/issues/34378)"""
```

python function comments are supposed to be placed right after the function definition, you can remove the following two lines.